src/utils/scoring/BestElementFinder.cpp

```diff
@@ -47,17 +49,32 @@ std::optional<size_t> cura::BestElementFinder::findBestElement(const size_t cand
                 iterator->score += weighed_criterion.criterion->computeScore(iterator->candidate_index) * weighed_criterion.weight;
             }
 
-            best_score = std::max(best_score, iterator->score);
+            if (best_candidate_iterator == end || iterator->score > best_candidate_iterator->score)
+            {
+                best_candidate_iterator = iterator;
+            }
+        }
+
+        if (best_candidate_iterator == end)
+        {
+            // Something got wrong, we have not best candidate
```

Comment:
I'd suggest changing this to "Something went wrong, we don't have the best candidate."